addons/mail/static/src/core/common/message_search_hook.js

```diff
@@ -72,17 +72,20 @@ export function useMessageSearch(thread) {
         async search(before = false) {
             if (this.searchTerm) {
                 this.searching = true;
-                const { count, loadMore, messages } = await sequential(() =>
+                const data = await sequential(() =>
                     threadService.search(this.searchTerm, this.thread, before)
                 );
-                this.searched = true;
-                this.searching = false;
-                this.count = count;
-                this.loadMore = loadMore;
-                if (before) {
-                    this.messages.push(...messages);
-                } else {
-                    this.messages = messages;
+                if (data) {
+                    const { count, loadMore, messages } = data;
+                    this.searched = true;
```

Comment:
Instead of validating positive result, you can validate negative result and return from there. It will reduce the diff.